crates/astria-sequencer/src/service/consensus.rs

```diff
@@ -140,11 +139,42 @@ impl Consensus {
         Ok(response::InitChain::default())
     }
 
+    fn prepare_proposal(
+        mut prepare_proposal: request::PrepareProposal,
+    ) -> anyhow::Result<response::PrepareProposal> {
+        let action_commitment = generate_transaction_commitment(&prepare_proposal.txs)
+            .context("failed to generate transaction commitment")?;
+        let mut txs: Vec<Bytes> = vec![action_commitment.to_vec().into()];
+        txs.append(&mut prepare_proposal.txs);
+        Ok(response::PrepareProposal {
+            txs,
+        })
+    }
+
+    fn process_proposal(process_proposal: &request::ProcessProposal) -> anyhow::Result<()> {
+        let received_action_commitment: [u8; 32] = process_proposal
+            .txs
+            .first()
+            .context("no transaction commitment in proposal")?
+            .to_vec()
+            .try_into()
+            .map_err(|_| anyhow!("transaction commitment must be 32 bytes"))?;
+        let expected_action_commitment =
+            generate_transaction_commitment(&process_proposal.txs[1..])
+                .context("failed to generate transaction commitment")?;
+        ensure!(
+            received_action_commitment == expected_action_commitment,
+            "transaction commitment does not match expected",
+        );
+
+        Ok(())
+    }
+
     #[instrument(skip(self))]
     async fn begin_block(
         &mut self,
         begin_block: request::BeginBlock,
-    ) -> Result<response::BeginBlock, BoxError> {
+    ) -> anyhow::Result<response::BeginBlock> {
```

Comment:
These changes seem to be unrelated?

Reply:
yeah it's just unifying the signature for all these methods